app/upgrades/v11/upgrades.go

```diff
@@ -74,3 +94,78 @@ func MigrateEscrowAccounts(ctx sdk.Context, ak authkeeper.AccountKeeper) {
 		ak.SetModuleAccount(ctx, acc)
 	}
 }
+
+// HandleRewardDistribution handles the logic for the reward distribution, it only commits to the db if successful
+func HandleRewardDistribution(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper, logger log.Logger) {
+	// use a cache context as a rollback mechanism in case
+	// the distrbution fails
+	cacheCtx, writeFn := ctx.CacheContext()
+	err := DistributeRewards(cacheCtx, bk, sk, dk)
+	if err != nil {
+		// log error instead of aborting the upgrade
+		logger.Error("failed to distribute rewards", "error", err.Error())
+	} else {
+		writeFn()
+	}
+}
+
+// DistributeRewards distributes the token allocations from the Olympus Mons incentivized testnet
+func DistributeRewards(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper) error {
+	funder := sdk.MustAccAddressFromBech32(FundingAccount)
+	numValidators := sdk.NewInt(int64(len(Validators)))
+
+	for _, allocation := range Allocations {
+
+		// send rewards to receivers
+		receiver := sdk.MustAccAddressFromBech32(allocation[0])
+		receivingAmount, ok := sdk.NewIntFromString(allocation[1])
+		if !ok {
+			return errorsmod.Wrapf(errortypes.ErrInvalidType,
+				"cannot retrieve allocation from string for address %s",
+				allocation[0])
+		}
+		reward := sdk.Coins{
```

Comment:
nit, could read nicer in one line
```suggestion
		reward := sdk.Coins{sdk.NewCoin(types.BaseDenom, receivingAmount)}
```